src/chigame/games/models.py

```diff
@@ -0,0 +1,62 @@
+from django.db import models
+
+from chigame.users.models import FriendGroup, User
+
+
+class Game(models.Model):
+    """
+    A game (Chess, Checkers, etc.)
+    """
+
+    name = models.TextField()
+    description = models.TextField()
+    min_players = models.PositiveIntegerField()
+    max_players = models.PositiveIntegerField()
+
+
+class Lobby(models.Model):
+    """
+    A lobby that users can join before starting a match.
+    The match can start as soon as min_players join the lobby.
+    """
+
+    name = models.TextField()
+    game = models.ForeignKey(Game, on_delete=models.CASCADE)
+    created_by = models.ForeignKey(User, on_delete=models.CASCADE)
+    members = models.ManyToManyField(User, related_name="lobbies")
+    min_players = models.PositiveIntegerField()
+    max_players = models.PositiveIntegerField()
+
+
+class Match(models.Model):
+    """
+    A specific match of a game, between a set of players.
+    """
+
+    game = models.ForeignKey(Game, on_delete=models.CASCADE)
+    lobby = models.OneToOneField(Lobby, on_delete=models.CASCADE)
+    winners = models.ManyToManyField(User, related_name="won_matches", blank=True)
+    players = models.ManyToManyField(User, through="Player")
+
+
+class Player(models.Model):
+    """
+    A player in a match.
+    """
+
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    match = models.ForeignKey(Match, on_delete=models.CASCADE)
```

Comment:
Why is this included as a through table? Does the definition of the ManytoMany field not implicitly define this table?

Is it to make it easier for the Games/matches team to flesh out the model?

Or it is necessary now

Reply:
It is very likely that Players will end up having attributes, so we may as well go ahead and define it explicitly as a through table.